batch_aggregator_proxy/batch_buffer.py

```diff
@@ -0,0 +1,65 @@
+import asyncio
+import logging
+from collections import defaultdict
+from urllib.parse import urljoin
+
+import httpx
+
+from batch_aggregator_proxy.schema import ProxyConfig
+
+
+class BatchBuffer:
+    def __init__(self, config: ProxyConfig, logger: logging.Logger):
+        """Initialize the buffer manager with the provided configuration."""
+        self.logger = logger
+        self.node_base_url = f"http://{config.NODE_HOST}:{config.NODE_PORT}"
+        self.flush_volume = config.FLUSH_THRESHOLD_VOLUME  # Max buffer size before flush
+        self.flush_interval = config.FLUSH_THRESHOLD_TIMEOUT  # Flush time threshold
+        self.buffer_queue = asyncio.Queue()
+        self._stop_event = asyncio.Event()
+        self._flush_task = asyncio.create_task(self._periodic_flush())
+
+    async def _periodic_flush(self):
+        """Periodically flushes the buffer based on the timeout interval."""
+        while not self._stop_event.is_set():
+            await asyncio.sleep(self.flush_interval)
+            await self.flush()
+
+    async def flush(self):
+        """Flush the buffer if it has items."""
+        if self.buffer_queue.empty():
+            return
+
+        batches_mapping = defaultdict(list)
+        batches_count = 0
+        while not self.buffer_queue.empty() and batches_count <= self.flush_volume:
+            app_name, batch = await self.buffer_queue.get()
```

Comment:
I suggest using get_nowait without await to ensure we do not switch between tasks while gathering batches to make things simpler.